diff -ruN Python-3.4.0.orig/Modules/pwdmodule.c Python-3.4.0/Modules/pwdmodule.c
--- Python-3.4.0.orig/Modules/pwdmodule.c	2014-03-17 02:31:31.000000000 +0000
+++ Python-3.4.0/Modules/pwdmodule.c	2014-07-06 17:46:02.000000000 +0100
@@ -72,7 +72,11 @@
     SETS(setIndex++, p->pw_passwd);
     PyStructSequence_SET_ITEM(v, setIndex++, _PyLong_FromUid(p->pw_uid));
     PyStructSequence_SET_ITEM(v, setIndex++, _PyLong_FromGid(p->pw_gid));
+#if !defined(HAVE_BROKEN_GECOS)
     SETS(setIndex++, p->pw_gecos);
+#else
+    SETS(setIndex++, "");
+#endif
     SETS(setIndex++, p->pw_dir);
     SETS(setIndex++, p->pw_shell);
 
diff -ruN Python-3.4.0.orig/Python/formatter_unicode.c Python-3.4.0/Python/formatter_unicode.c
--- Python-3.4.0.orig/Python/formatter_unicode.c	2014-03-17 02:31:31.000000000 +0000
+++ Python-3.4.0/Python/formatter_unicode.c	2014-07-06 17:47:12.000000000 +0100
@@ -664,6 +664,7 @@
 get_locale_info(int type, LocaleInfo *locale_info)
 {
     switch (type) {
+#if !defined(HAVE_BROKEN_LOCALECONV)
     case LT_CURRENT_LOCALE: {
         struct lconv *locale_data = localeconv();
         locale_info->decimal_point = PyUnicode_DecodeLocale(
@@ -681,6 +682,9 @@
         locale_info->grouping = locale_data->grouping;
         break;
     }
+#else
+    case LT_CURRENT_LOCALE:
+#endif
     case LT_DEFAULT_LOCALE:
         locale_info->decimal_point = PyUnicode_FromOrdinal('.');
         locale_info->thousands_sep = PyUnicode_FromOrdinal(',');
diff -ruN Python-3.4.0.orig/Python/pystrtod.c Python-3.4.0/Python/pystrtod.c
--- Python-3.4.0.orig/Python/pystrtod.c	2014-03-17 02:31:31.000000000 +0000
+++ Python-3.4.0/Python/pystrtod.c	2014-07-06 17:53:02.000000000 +0100
@@ -165,7 +165,9 @@
 {
     char *fail_pos;
     double val;
+#if !defined(HAVE_BROKEN_LOCALECONV)
     struct lconv *locale_data;
+#endif
     const char *decimal_point;
     size_t decimal_point_len;
     const char *p, *decimal_point_pos;
@@ -177,8 +179,12 @@
 
     fail_pos = NULL;
 
+#if !defined(HAVE_BROKEN_LOCALECONV)
     locale_data = localeconv();
     decimal_point = locale_data->decimal_point;
+#else
+    decimal_point = ".";
+#endif
     decimal_point_len = strlen(decimal_point);
 
     assert(decimal_point_len != 0);
@@ -378,6 +384,7 @@
 Py_LOCAL_INLINE(void)
 change_decimal_from_locale_to_dot(char* buffer)
 {
+#if !defined(HAVE_BROKEN_LOCALECONV)
     struct lconv *locale_data = localeconv();
     const char *decimal_point = locale_data->decimal_point;
 
@@ -402,6 +409,7 @@
             }
         }
     }
+#endif
 }
 
 
